README.md

```diff
@@ -16,3 +16,9 @@ See [the Playwright command line docs](https://playwright.dev/docs/test-cli) for
 From the project directory run `ng serve` to spin up the web server and navigate to <http://localhost:4200/> - you don't need to do this before running tests as that's handled by the Playwright Test framework.
 
 The app will automatically reload if you change any of the source files.
+
+## Pipeline
+
+The Azure Devops pipeline is configured to install prerequisites, run the tests, publish test results and then publish test result attachments as a pipeline artifact.
+
+When a test fails in the pipeline it is retried and a trace is recorded of the retry - this trace can be analysed to see the entire flow of the test including the full dom before and after each Playwright action. See [Playwright Trace docs](https://playwright.dev/docs/trace-viewer) for information on how to view the trace.
```

Comment:
do we have a number of retries that the pipeline does before it fails, think it does but i'm not an azure pipeline expert.

Reply:
It is currently configured in playwright.config.ts line 27. There are many playwright options to control retries but I'm not fond of retrying UI tests. The only reason I'm including a retry here is for the trace recording